File: python/src/curious/commands/roadmap.py
from __future__ import annotations

import sys
from pathlib import Path

from curious.config import resolve_config
from curious.harness import run_harness
from curious.prompts_tasks import build_roadmap_prompt


def run_roadmap(config_path: str | None, verbose: bool) -> None:
    config = resolve_config(config_path=config_path, require_spec=True)
    spec_body = Path(config.spec_path).read_text(encoding="utf-8")
    prompt = build_roadmap_prompt(config, spec_body)
    result = run_harness(
        prompt,
        Path(config.cwd),
        config.llm,
        config.harness,
        verbose=verbose,
    )

    if result.status != "finished":
        print(f"[curious] roadmap failed: {result.error}")
        sys.exit(1)

    print("\n[curious] Next: curious-py run --cycle")

File: src/commands/roadmap.ts
import { readFile } from "node:fs/promises";
import { resolveConfig } from "../config.js";
import { runAgentTask } from "../agent-run.js";
import { buildRoadmapPrompt } from "../prompts-tasks.js";

export async function runRoadmap(
  configPath: string | undefined,
  verbose: boolean,
): Promise<void> {
  const config = await resolveConfig({ configPath, requireSpec: true });
  const specBody = await readFile(config.specPath, "utf8");

  const prompt = await buildRoadmapPrompt(config, specBody);
  const result = await runAgentTask(config, prompt, {
    verbose,
    label: "roadmap",
  });

  if (result.status !== "finished") {
    process.exit(1);
  }

  console.log("\n[curious] Next steps:");
  console.log("  1. Review Roadmap and Progress in spec/SPEC.md");
  console.log("  2. curious run --cycle   # develop → review → sync");
}
